pkgs/applications/video/giph/default.nix

```diff
@@ -0,0 +1,63 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, ffmpeg
+, xdotool
+, withSlop ? true
+, slop
+, withNotify ? true
+, libnotify
+, withPgrep ? true
+, procps
+}:
+
+stdenv.mkDerivation rec {
+  pname = "giph";
+  version = "1.1.1";
+
+  src = fetchFromGitHub {
+    owner = "phisch";
+    repo = "giph";
+    rev = version;
+    sha256 = "19l46m1f32b3bagzrhaqsfnl5n3wbrmg3sdy6fdss4y1yf6nqayk";
+  };
+
+  postPatch = ''
+    substituteInPlace src/giph \
+      --replace '=(ffmpeg' '=(${ffmpeg}/bin/ffmpeg' \
+      --replace '="$(xdotool' '="$(${xdotool}/bin/xdotool' \
+  '' + lib.optionalString withSlop ''
+      --replace '=(ffmpeg' '=(${ffmpeg}/bin/ffmpeg' \
+  '' + lib.optionalString withNotify ''
+      --replace '=(notify-send' '=(${libnotify}/bin/notify-send' \
+  '' + lib.optionalString withPgrep ''
+      --replace '"$(pgrep' '"$(${procps}/bin/pgrep'
+  '';
+
+  buildInputs = [
+    ffmpeg
+    xdotool
+  ]
+  ++ lib.optional
+    withSlop
+    slop
+  ++ lib.optional
+    withNotify
+    libnotify
+  ++ lib.optional
+    withPgrep
+    procps;
```

Comment:
shell scripts don't need to declare buildInputs for the executables they use at runtime. see also: https://github.com/NixOS/nixpkgs/pull/117348 and  [Nix manual "complete dependencies"](https://nixos.org/manual/nix/stable/#idm140737322654464)

Reply:
Baesd on your PR, I just shouldn't specify them at all in the buildInputs?
Only import them in the first line?

Reply:
They don't need to be at all in buildInputs. That is only required for autopatchelfHook